src/Diagnostics.DataProviders/DataProviders/KustoDataProvider.cs

```diff
@@ -39,8 +41,8 @@ public async Task<DataTable> ExecuteClusterQuery(string query, string requestId
 
         public async Task<DataTable> ExecuteQuery(string query, string stampName, string requestId = null, string operationName = null)
         {
-            await AddQueryInformationToMetadata(query, stampName);
             var cluster = GetClusterNameFromStamp(stampName);
+            await AddQueryInformationToMetadata(query, cluster);
             return await _kustoClient.ExecuteQueryAsync(query, cluster, _configuration.DBName, requestId, operationName);
```

Comment:
Can you log additional message that states the cluster that were about to send a request to?